### DISCUSSION TITLE
Storybook code coverage in Turbo Monorepo.

### DISCUSSION THREAD

**Discussion by votemike** [storybookjs/storybook#25820] [Help]
### Summary

[I have created a new Turbo monorepo](https://github.com/votemike/design-system/commits/main/) using [the design-system example](https://github.com/vercel/turbo/tree/main/examples/design-system).  
I then added testing to Storybook following [these instructions](https://storybook.js.org/docs/writing-tests/test-runner). In the root of the repo, it is now possible to run `pnpm run test-storybook` (assuming storybook is running).  
I then [added test coverage](https://github.com/votemike/design-system/commit/a05a5479dd39ee424e9b5580de0d7d87fcc05ada) as per [this](https://storybook.js.org/docs/writing-tests/test-coverage) which I was hoping would show me uncovered lines/files in the packages/ui/src directory by running `pnpm run test-storybook -- --coverage`, but I can't get it to do so. I get a single line called "All files" which only shows zeros.

<img width="822" alt="Screenshot 2024-01-30 at 14 18 15" src="https://github.com/vercel/turbo/assets/3957065/31a3169b-8129-4aff-b60f-f4b483762124">

Is this possible? Is it a limitation of Turbo/Storybook in some way? Can this be done in monorepos? Is it just not documented how to do such a thing?  
What am I doing wrong?

Thanks in advance.

### Additional information

```
import { dirname, join, resolve } from "path";

function getAbsolutePath(value) {
  return dirname(require.resolve(join(value, "package.json")));
}

const config = {
  stories: ["../stories/*.stories.tsx", "../stories/**/*.stories.tsx"],
  addons: [
    getAbsolutePath("@storybook/addon-links"),
    getAbsolutePath("@storybook/addon-essentials"),
    getAbsolutePath("@storybook/addon-docs"),
    getAbsolutePath("@storybook/addon-interactions"),
    {
      name: getAbsolutePath('@storybook/addon-coverage'),
      options: {
        istanbul: {
          include: ['../../../packages/ui/src/**'],
          debug: true,
        },
      },
    },
  ],
  framework: {
    name: getAbsolutePath("@storybook/react-vite"),
    options: {},
  },

  core: {},

  async viteFinal(config, { configType }) {
    // customize the Vite config here
    return {
      ...config,
      define: { "process.env": {} },
      resolve: {
        alias: [
          {
            find: "ui",
            replacement: resolve(__dirname, "../../../packages/ui/"),
          },
        ],
      },
    };
  },

  docs: {
    autodocs: true,
  },
};

export default config;
```

### Create a reproduction

https://github.com/votemike/design-system

**Comment by haileypun** [comment]
I'm struggling to do the same despite the file structures being defined in the config. Would be great to get some support on this!

**Comment by woodsboe** [comment]
Hi @votemike and @haileypun,
Could you try adding the UI package path to the stories property like this?
```typescript
{
  stories: ["../stories/*.stories.tsx", "../stories/**/*.stories.tsx", "../../../packages/ui/**/*/*.stories.tsx"],
  ...
  }
```

**Reply by votemike** [reply]
I have tried that, and it also doesn't work.

**Comment by lenniii** [comment]
@votemike @haileypun did you resolve the issue by any chance?

**Reply by votemike** [reply]
No. Unfortunately not. We still have no coverage.

**Comment by ElBartoTn** [comment]
I used  "vite-plugin-istanbul" with vite config to make it works , i used this vite.config.ts file 
```
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";
import svgr from "vite-plugin-svgr";
import istanbul from "vite-plugin-istanbul";

export default defineConfig({
  plugins: [
    react(),
    svgr(),
    istanbul({
      include: "**/src/**",
      exclude: ["node_modules", "test/"],
      extension: [".js", ".ts", ".tsx"],
      requireEnv: false,
    }),
  ],
  server: {
    watch: {
      ignored: ["**/coverage/**", "**/node_modules/**", "**/icons/**"],
    },
  },
  build: {
    cssCodeSplit: true,
  },
});
```
I removed the option from code-coverage addon

**Reply by nutrian** [reply]
@ElBartoTn is that vite config from your `apps/storybook` or `packages/ui` ?

**Reply by ElBartoTn** [reply]
Its storybook
